presto-parser/src/main/java/com/facebook/presto/sql/parser/SqlParser.java

```diff
@@ -63,17 +63,35 @@ public SqlParser(SqlParserOptions options)
         allowedIdentifierSymbols = EnumSet.copyOf(options.getAllowedIdentifierSymbols());
     }
 
+    /**
+     * Consider using {@link #createStatement(String, ParsingOptions)}
+     */
+    @Deprecated
```

Comment:
Why?

Reply:
This method became incapable of parsing decimal literals (those without type explicitly specified), as `ParsingOptions` default to rejecting ambiguities.